include/libp2p/outcome/outcome.hpp

```diff
@@ -10,7 +10,15 @@
 #include <boost/outcome/success_failure.hpp>
 #include <boost/outcome/try.hpp>
 
-#define OUTCOME_TRY(...) BOOST_OUTCOME_TRY(__VA_ARGS__)
+// To define OUTCOME_TRY macro, we will need to create OUTCOME_TRY_1 and
+// OUTCOME_TRY_2 depending on number of arguments
+#define OUTCOME_TRY_1(...) BOOST_OUTCOME_TRY(__VA_ARGS__)
+#define OUTCOME_TRY_2(...) BOOST_OUTCOME_TRY(auto __VA_ARGS__)
+
+// trick from https://stackoverflow.com/a/11763277 to overload OUTCOME_TRY
+#define GET_MACRO(_1, _2, NAME, ...) NAME
+#define OUTCOME_TRY(...) \
+  GET_MACRO(__VA_ARGS__, OUTCOME_TRY_2, OUTCOME_TRY_1)(__VA_ARGS__)
```

Comment:
Was needed, because newer boost required `auto` before first argument to BOOST_OUTCOME_TRY (if there are two args)

Reply:
Should we write `BOOST_OUTCOME_TRY(l, r)` when refactoring existing code?
E.g. unwrapping directly to member `BOOST_OUTCOME_TRY(some.member, some_result)`